app/models/periodic_expense.rb

```diff
@@ -0,0 +1,50 @@
+class PeriodicExpense < ActiveRecord::Base
+  PERIODS = { 'weekly' => 1.week, 'monthly' => 1.month }
+
+  belongs_to :user
+
+  validates :name, :user, :amount, :period, :start_date, presence: true
+  validates :period, inclusion: { in: PERIODS.keys }
+  validate :end_date_is_after_start_date
+  validate :last_paid_at_is_between_start_and_end_dates
+
+  def period_as_time
+    PERIODS[period]
+  end
+
+  def ready_to_pay?
+    !next_pay.future?
```

Comment:
those could also change to something like started_on and ended_on, but since they already have `date` in the name, I don't see that as much of a problem. They're fine that way as well